projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionService.java

```diff
@@ -273,7 +273,31 @@ public DataJobExecution readJobExecution(String teamName, String jobName, String
               .lastDeployedDate(jobExecution.getDeployedDate())
               .lastDeployedBy(jobExecution.getDeployedBy())
               .build();
-      return Optional.of(jobExecutionRepository.save(dataJobExecution));
+      jobExecutionRepository.save(dataJobExecution);
+   }
+
+   /**
+    * Returns the last execution of the data job with the specified name, or an empty optional if there
+    * are no executions. The last execution is considered the one with the most recent start time.
+    *
+    * @param dataJobName The name of the data job whose last execution to return.
+    * @return The last execution of the data job if any, otherwise, {@link Optional#empty()}.
+    */
+   public Optional<com.vmware.taurus.service.model.DataJobExecution> getLastExecution(String dataJobName) {
```

Comment:
is this method used?

Reply:
No, I was planning to use it later, but its better to introduce it in the commit where it will be used, so I will remove it for now.